gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -204,7 +205,10 @@ case class TransformPreOverrides(isAdaptiveContext: Boolean)
           val pos = newExpressionsPosition(i)
           newExprs = newExprs :+ project.output(pos)
         }
-        (projectExpressions.size, HashPartitioning(newExprs, numPartitions), project)
+        (
+          projectExpressions.size,
+          new HashPartitioningWrapper(exprs, newExprs, numPartitions),
```

Comment:
I'm not sure I get it. AQE won't touch the expressions in hash partitioning, then how can it affect AQE behavior ? Does it also affect range partitioning?

Reply:
https://github.com/apache/spark/blob/b9cc813adf0ba592f020bc9778b373800ad81b8f/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala#L180C30-L180C30 here `ValidateRequirements.validate(applied, distribution)` will validate failed due to shuffle hash expressions changed.

Reply:
I see, and these codes only be used in ch backend so Velox backend does not have this issue.

Reply:
looks reasonable to me